src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -9,59 +9,30 @@ namespace System.Threading
-                            {
-                                return defaultValue;
-                            }
-                            return result;
-                        }
-                    case IConvertible convertible:
-                        return convertible.ToInt32(NumberFormatInfo.InvariantInfo);
-                }
-
-                return defaultValue;
-            }
-
             private static HillClimbing CreateHillClimber()
             {
                 // Default values pulled from CoreCLR
-                return new HillClimbing(wavePeriod: GetConfig("HillClimbing_WavePeriod", 4),
-                    maxWaveMagnitude: GetConfig("HillClimbing_MaxWaveMagnitude", 20),
-                    waveMagnitudeMultiplier: GetConfig("HillClimbing_WaveMagnitudeMultiplier", 100) / 100.0,
-                    waveHistorySize: GetConfig("HillClimbing_WaveHistorySize", 8),
-                    targetThroughputRatio: GetConfig("HillClimbing_Bias", 15) / 100.0,
-                    targetSignalToNoiseRatio: GetConfig("HillClimbing_TargetSignalToNoiseRatio", 300) / 100.0,
-                    maxChangePerSecond: GetConfig("HillClimbing_MaxChangePerSecond", 4),
-                    maxChangePerSample: GetConfig("HillClimbing_MaxChangePerSample", 20),
-                    sampleIntervalMsLow: GetConfig("HillClimbing_SampleIntervalLow", 10),
-                    sampleIntervalMsHigh: GetConfig("HillClimbing_SampleIntervalHigh", 200),
-                    errorSmoothingFactor: GetConfig("HillClimbing_ErrorSmoothingFactor", 1) / 100.0,
-                    gainExponent: GetConfig("HillClimbing_GainExponent", 200) / 100.0,
-                    maxSampleError: GetConfig("HillClimbing_MaxSampleErrorPercent", 15) / 100.0
+                return new HillClimbing(wavePeriod: AppContextConfigHelper.GetConfig("HillClimbing_WavePeriod", 4),
+                    maxWaveMagnitude: AppContextConfigHelper.GetConfig("HillClimbing_MaxWaveMagnitude", 20),
+                    waveMagnitudeMultiplier: AppContextConfigHelper.GetConfig("HillClimbing_WaveMagnitudeMultiplier", 100) / 100.0,
+                    waveHistorySize: AppContextConfigHelper.GetConfig("HillClimbing_WaveHistorySize", 8),
+                    targetThroughputRatio: AppContextConfigHelper.GetConfig("HillClimbing_Bias", 15) / 100.0,
+                    targetSignalToNoiseRatio: AppContextConfigHelper.GetConfig("HillClimbing_TargetSignalToNoiseRatio", 300) / 100.0,
+                    maxChangePerSecond: AppContextConfigHelper.GetConfig("HillClimbing_MaxChangePerSecond", 4),
+                    maxChangePerSample: AppContextConfigHelper.GetConfig("HillClimbing_MaxChangePerSample", 20),
+                    sampleIntervalMsLow: AppContextConfigHelper.GetConfig("HillClimbing_SampleIntervalLow", 10),
+                    sampleIntervalMsHigh: AppContextConfigHelper.GetConfig("HillClimbing_SampleIntervalHigh", 200),
```

Comment:
These two values are unsigned in CoreCLR. With them being signed here, you should probably reject negative values and use the default. You should also reject both values if min > max and use defaults.

In this case it's probably ok to parse these as signed even though the value is unsigned, but generally for values that are not expected to be negative and may have the full uint range, it would be better to parse them as unsigned and not allow negative sign.